src/Routes/ListRoutesCommand.php

```diff
@@ -0,0 +1,231 @@
+        supported by the preg_* functions. For example, 
+        "\Mezzio\Middleware\LazyLoadingMiddleware,LazyLoadingMiddleware,\Mezzio*".
+        EOT;
+
+    private const HELP_OPT_HAS_NAME = <<<'EOT'
+        Filters out routes by name. This option accepts a comma-separated list of 
+        one or more names. The names can be fixed-strings or regular expressions 
+        supported by the preg_* functions. For example, 
+        "user,user.register,*.register,user*".
+        EOT;
+
+    private const HELP_OPT_HAS_PATH = <<<'EOT'
+        Filter out routes by path. This option accepts a comma-separated list of 
+        one or more paths. The paths can be a fixed-string or a regular expression, 
+        supported by the preg_* functions. For example, "/,/api/ping,*/ping".
+        EOT;
+
+    private const HELP_OPT_SORT = <<<'EOT'
+        Sort the command's output. The supported values are "name" and "path".
+        EOT;
+
+    private const HELP_OPT_SUPPORTS_METHOD = <<<'EOT'
+        Filters out routes by HTTP method.  This option accepts a comma-separated 
+        list of one or more HTTP methods.
+        EOT;
+
+    /** @var null|string Cannot be defined explicitly due to parent class */
+    public static $defaultName = 'mezzio:routes:list';
+
+    public function __construct(RouteCollector $routeCollector)
+    {
+        $this->routes = $routeCollector->getRoutes();
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription("Print the application's routing table.");
+        $this->setHelp(self::HELP);
+
+        $this->addOption(
+            'format',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_FORMAT,
+            'table'
+        );
+
+        $this->addOption(
+            'sort',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_SORT,
+            'name'
+        );
+
+        // Routing table filter options
+        $this->addOption(
+            'has-middleware',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_MIDDLEWARE,
+            false
+        );
+        $this->addOption(
+            'has-name',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_NAME,
+            false
+        );
+        $this->addOption(
+            'has-path',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_PATH,
+            false
+        );
+        $this->addOption(
+            'supports-method',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_SUPPORTS_METHOD,
+            false
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $result = 0;
+        if (empty($this->routes)) {
```

Comment:
Since the `routes` property is an array, `count($this-routes) === 0` should be used instead to show clear intend. The use of `empty` is oftentimes confusing or misleading.

Reply:
Or `[] === $var` :+1: